### PR TITLE
Companion for substrate#14435 (BenchmarkHelper for frame-system)

### PR DESCRIPTION
**Pull request by bkontur**
https://github.com/paritytech/substrate/pull/14435

Relates to: https://github.com/paritytech/cumulus/issues/2765#issuecomment-1600779522

### CODE REVIEW

**Comment by bkontur** [comment]
bot bench $ pallet asset-hub-kusama assets frame_system

**Comment by bkontur** [comment]
bot bench $ pallet asset-hub-kusama assets frame_system

**Review by acatangiu — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

**Review by joepetrowski — APPROVED** [review]

File: pallets/parachain-system/src/lib.rs [hunk 1]
```diff
@@ -1067,6 +1067,33 @@ impl<T: Config> Pallet<T> {
...
+		let vfp = PersistedValidationData {
+			parent_head: polkadot_parachain::primitives::HeadData(Default::default()),
+			relay_parent_number: 1,
+			relay_parent_storage_root: Default::default(),
+			max_pov_size: 1_000,
+		};
+		<ValidationData<T>>::put(&vfp);
+
+		// insert dummy HostConfiguration with
+		let host_config = AbridgedHostConfiguration {
```
Comment (rphmeier) [inline pallets/parachain-system/src/lib.rs:1084]: Setting all values here to zero may lead to unexpected behavior in the benchmark, if e.g. `set_validation_data` is called within the benchmark.

**Review by bkchr — APPROVED** [review]

File: pallets/parachain-system/src/lib.rs (hunk 1 above)
Comment (bkontur) [reply pallets/parachain-system/src/lib.rs:1084]: ok, thank you, changed to non-zero values

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-43/3158/4

**Pull request merged**